cranelift-codegen/src/write.rs

```diff
@@ -510,6 +505,20 @@ pub fn write_operands(
         NullAry { .. } => write!(w, " "),
         InsertLane { lane, args, .. } => write!(w, " {}, {}, {}", args[0], lane, args[1]),
         ExtractLane { lane, arg, .. } => write!(w, " {}, {}", arg, lane),
+        Const {
+            constant_handle, ..
+        } => {
+            let data = dfg.constants.get(constant_handle);
+            let uimm128 = Uimm128::from(&data[..]);
```

Comment:
This is a good suggestion. I only used `Uimm128` because it had the `Display` and `FromStr` implementations needed to make this work. However, looking at it now, if those implementations were on `ConstantData` (the result type of `dfg.constants.get(constant_handle)`) then we might be able to be more flexible. The issue with that is that currently `ConstantData` is a type alias and cannot yet implement anything. I suggest that we do this as a separate issue: convert `ConstantData` to a real type, add appropriate `From`, `Display` and `FromStr` implementations, and then use it directly here instead of `Uimm128`. Thoughts?

Reply:
Opened this: https://github.com/CraneStation/cranelift/issues/1027.